### PR TITLE
Spam revisions page fixed

### LINKED ISSUE #3082: spam/revisions page down
Hi!
Looks like the https://publiclab.org/spam/revisions page is down. Any ideas? 

Best,
Stevie

### PR DESCRIPTION
**Pull request by gauravano**
Fixes #3082 

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] PR is descriptively titled 📑
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [x] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

File: app/models/node.rb [hunk 1]
```diff
@@ -71,7 +71,7 @@ def updated_month
     updated_at.strftime('%B %Y')
   end
 
-  has_many :revision, foreign_key: 'nid' # , dependent: :destroy # re-enable in Rails 5
```
Comment (gauravano) [inline app/models/node.rb]: Hi @jywarren, I just found this `dependent : : destroy` commented due to which if a node of spammed revision is deleted by author or  moderator, the revisions aren't deleted. 

And, ` # , dependent: :destroy # re-enable in Rails 5` is present at many places so we have collecting redundant data lately, which is still present. 

Should we enable all `dependent: :destroy`?

**Comment by gauravano** [comment]
@jywarren fix is working! Verified from unstable. Please review before merging. Thanks!

File: app/models/node.rb (hunk 1 above)
Comment (gauravano) [reply app/models/node.rb]: Hey @Souravirus any idea about these comments? Thanks!

Comment (Souravirus) [reply app/models/node.rb]: No, I didn't had any idea about these. Sorry about that.

Comment (gauravano) [reply app/models/node.rb]: No issue!

Comment (gauravano) [reply app/models/node.rb]: @jywarren these all comments are put by you at Rails 4.1 upgrade - https://github.com/publiclab/plots2/pull/1799/ 

Do you think we can remove these comments now? Also, should we delete the dependent data like revisions without node, etc?

Comment (jywarren) [reply app/models/node.rb]: Hmm, i think we could open an issue for this - It was so long ago i don't remember. I think we have to be very very careful here, as incorrect `dependent: destroy` has caused data loss in the past. Let's do it separate from this PR! Thanks!

Comment (jywarren) [reply app/models/node.rb]: image

File: app/models/node.rb [hunk 2]
```diff
@@ -71,7 +71,7 @@ def updated_month
...
+  has_many :revision, foreign_key: 'nid', dependent: :destroy
```
Comment (gauravano) [inline app/models/node.rb]: @jywarren but here :arrow_up: I have added this `dependent: :destroy`. Should I remove this?

Comment (jywarren) [reply app/models/node.rb]: uh oh - oops, i missed this! Um, I think this is OK. I read the docs carefully. Thanks!

Comment (gauravano) [reply app/models/node.rb]: I also peeked :mag: at this :arrow_right: https://guides.rubyonrails.org/association_basics.html while changing so this one should be fine.

**Pull request merged**